pkg/reconciler/common/releases.go

```diff
@@ -82,6 +82,12 @@ func InstalledManifest(instance v1alpha1.KComponent) (mf.Manifest, error) {
 func IsVersionValidMigrationEligible(instance v1alpha1.KComponent) error {
 	var err error
 	target := sanitizeSemver(TargetVersion(instance))
+	if target == "v" {
+		// If the function TargetVersion returns empty string, if means the specified target version is not available
+		// in this release to install.
```

Comment:
This doesn't seem to be true, by looking at the code, it seems like `TargetVersion()` returns either:
* the latest available release
* `instance.GetSpec().GetVersion()`

If the version is not found, it returns `instance.GetSpec().GetVersion()`. Not `""`.

Could you please clarify the intention?

Reply:
If `spec.version` is specified, but not available in the supported list of version numbers, TargetVersion returns "".
If `spec.version` is specified, and available in the supported list of version numbers, TargetVersion returns spec.version with the latest patch number.
If `spec.version` is NOT specified, TargetVersion returns the latest release number.

Reply:
Ok, I see the behavior now, somewhere deep inside `getLatestRelease()` (which name is misleading). IMO that code is very convoluted and should return an error if it can't return an actual version, but that's a topic for its own issue I think.

Reply:
@antoineco I would like to change the logic for getLatestRelease.
There is an inconsistency between major.minor and major.minor.patch formats, if they are not found.
If getLatestRelease still returns the spec.version, no matter they are available or not. We can catch the error later, when parsing manifests.